lib/krane/kubernetes_resource/deployment.rb

```diff
@@ -91,6 +91,11 @@ def deploy_timed_out?
       return false if deploy_failed?
       return super if timeout_override
 
+      if progress_condition.present?
+        StatsD.client.increment('kubectl.error', 1, tags: { context: context, namespace: namespace,
+                                                            progress_condition: deploy_failing_to_progress? })
+      end
```

Comment:
I don't really like the idea of instrumenting the resource models, themselves, and would prefer if we could move that to the actual task runner, where such instrumentation makes more sense. It's worth noting we already capture timeout errors and publish them via StatsD (see https://shopify.datadoghq.com/dashboard/5kc-557-amd/krane--kubernetes-deploy-dash?from_ts=1605624831373&live=true&to_ts=1605628431373).

Alternatively, since we are constrained by the `KubernetesResource` interface, could we add a `statsd_tag` in `Deployment#deploy_timed_out?` when `deploy_failing_to_progress?` is true to indicate an actual progressing failure? 🤔 It would be hard to know if it's a fail-fast or an initial progressing failure, though.